proposals/readonly-struct-init.md

```diff
@@ -0,0 +1,161 @@
+# readonly object initializers 
+
+* [x] Proposed
```

Comment:
Nit: I suggest to move the checklist to the championed issue. Otherwise, it never gets updated.

Reply:
Will do.